src/Utilities/cachingoptimizer.jl

```diff
@@ -306,7 +280,14 @@ MOI.is_empty(m::CachingOptimizer) = MOI.is_empty(m.model_cache)
 
 function MOI.optimize!(m::CachingOptimizer)
     if m.mode == AUTOMATIC && m.state == EMPTY_OPTIMIZER
-        _attach_optimizer(m, MOI.optimize_model!)
+        indexmap, copied = MOI.optimize!(m.optimizer, m.model_cache)
+        if copied
+            m.state = ATTACHED_OPTIMIZER
+        end
+        # MOI does not define the type of index_map, so we have to convert it
+        # into an actual IndexMap. Also load the reverse IndexMap.
```

Comment:
It does now: https://github.com/jump-dev/MathOptInterface.jl/blob/ddbb343858245cf562da4edfc921594a734c8900/src/MathOptInterface.jl#L113

Reply:
This is copy-paste from attach_optimizer. Maybe do that in another PR?